src/parser.js

```diff
@@ -386,24 +437,114 @@ export default class Parser extends Stream {
             },
             'skip'() {
               this.manifest.skip = entry.attributes;
+
+              if (!entry.attributes.hasOwnProperty('SKIPPED-SEGMENTS')) {
+                this.trigger('warn', {
+                  message: '#EXT-X-SKIP lacks required attribute SKIPPED-SEGMENTS'
+                });
+              }
             },
             'part'() {
               this.manifest.parts = this.manifest.parts || [];
               this.manifest.parts.push(entry.attributes);
+              const missingAttributes = [];
+
+              ['URI', 'DURATION'].forEach(function(k) {
+                if (!entry.attributes[k]) {
```

Comment:
if a duration is zero, this could theoretically fail, no? Might be better to use `k in entry.attributes`?